resources/assets/js/components/Registration/GovDeptAndEmailInputs.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from "react";
+import { InjectedIntlProps, injectIntl, defineMessages } from "react-intl";
+import ReactDOM from "react-dom";
+import { Department } from "../../models/types";
+import Select, { SelectOption } from "../Select";
+import IntlContainer from "../../IntlContainer";
+import Input from "../Input";
+
+interface GovDeptAndEmailInputsProps {
+  departments: Department[];
+  prevDeptId?: number;
+  prevGovEmail?: string;
+}
+
+const messages = defineMessages({
+  notInGov: {
+    id: "govDeptAndEmail.notInGov",
+    defaultMessage: "Not in Government",
+    description:
+      "The user's option to state that they don't have a government email.",
+  },
+  department: {
+    id: "govDeptAndEmail.departmentLabel",
+    defaultMessage: "Government Department",
+    description: "Label for the Government Department input field.",
+  },
+  departmentNullSelection: {
+    id: "govDeptAndEmail.departmentNullSelection",
+    defaultMessage: "Select a Department",
+    description:
+      "Null selection option for the Government Department input field.",
+  },
+  govEmail: {
+    id: "govDeptAndEmail.govEmailLabel",
+    defaultMessage: "Your Government Email",
+    description: "Label for government email input field.",
+  },
+});
+
+export const GovDeptAndEmailInputs: React.FunctionComponent<
+  GovDeptAndEmailInputsProps & InjectedIntlProps
+> = ({ departments, prevDeptId, prevGovEmail, intl }): React.ReactElement => {
+  const { locale } = intl;
+  if (locale !== "en" && locale !== "fr") {
+    throw new Error("Unknown intl.locale");
+  }
+
+  const [deptId, setDeptId] = useState<number | null>(
+    prevDeptId !== undefined ? prevDeptId : null,
+  );
+  const [govEmail, setGovEmail] = useState(prevGovEmail || "");
+
+  const NOT_IN_GOV = 0;
+  const notInGovOption = {
+    value: NOT_IN_GOV,
+    label: intl.formatMessage(messages.notInGov),
+  };
+  const options = [
+    notInGovOption,
+    ...departments.map(
+      (department): SelectOption => ({
+        value: department.id,
+        label: department[locale].name,
+      }),
+    ),
+  ];
+
+  const getEventDeptId = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ): number | null =>
+    event.target.value !== "" ? Number(event.target.value) : null;
+  return (
+    <>
+      <Select
```

Comment:
Curious why we're using a React component for a dropdown and an input? Seems like a lot of code, and potentially a degraded user experience, for a couple fields on a form.

Reply:
Hmm yeah this one is an interesting case. Toggling a class in vanilla JS is pretty trivial, but if we're committing to use React whenever we need a JS interaction it probably makes more sense to refactor the entire component/form/whatever in React, rather than having one or two fields within a form being React. I can see the potential for weird race condition bugs or less than ideal user experience with something like this.

Reply:
You're right. I jumped straight to react because I'm scared of the mess that is app.js, but I tried making a standalone script for this and it actually feels cleaner.  2ddb0543342308318a7a0cd2c55c6a1772caa16c
Thanks.